python_modules/libraries/dagster-dg/dagster_dg/cli/list.py

```diff
@@ -251,6 +254,30 @@ def _get_sensors_table(sensors: Sequence[DgSensorMetadata]) -> Table:
     return table
 
 
+# On older versions of `dagster`, `dagster-components list defs` output was written directly to
+# stdout, where it was possibly polluted by other output from user code. This scans raw stdout for
+# the line containing the output.
+def _extract_list_defs_output_from_raw_output(raw_output: str) -> list[Any]:
+    last_decode_error = None
+    for line in raw_output.splitlines():
+        try:
+            defs_list = deserialize_value(line, as_type=list[DgDefinitionMetadata])
+            return defs_list
+        except (json.JSONDecodeError, DeserializationError) as e:
+            last_decode_error = e
+
+    if last_decode_error:
+        raise last_decode_error
+
+    raise Exception(
+        "Did not successfully parse definitions list. Full stdout of subprocess:\n" + raw_output
+    )
+
+
+MIN_DAGSTER_COMPONENTS_LIST_DEFINITIONS_LOCATION_OPTION_VERSION = Version("1.10.8")
+MIN_DAGSTER_COMPONENTS_LIST_DEFINITIONS_OUTPUT_FILE_OPTION_VERSION = Version("1.10.12")
```

Comment:
this is pretty brutal - maybe CLI commands should expose some way to get their available outputs or something so we could make this more semantic vs. hard-coding specific versions?